src/tests/Microsoft.PowerFx.Json.Tests/ParseJSONTests.cs

```diff
@@ -547,6 +548,39 @@ public void FromJsonHeterogeneousPrimitiveArray(string json, string expectedType
                         break;
                 }
             }
+        }
+
+        [Theory]
+        [InlineData("[{\"Name\": \"Peter\"}]", "[]")]
+        [InlineData("[1,2,3]", "[]")]
+        [InlineData("[1,2,3]", "[5, 6]")]
+        [InlineData("[\"A\", \"B\"]", "[\"1\"]")]
+        [InlineData("[{\"Name\": \"Peter\"}]", "[{\"Name\": \"Julia\"}]")]
+        [InlineData("[]", "[]")]
+        public void FromArrayWithType(string prototypeJson, string valueJson)
+        {
+            FormulaValue prototypeValue = FormulaValueJSON.FromJson(prototypeJson);
```

Comment:
>FormulaValue prototypeValue = FormulaValueJSON.FromJson(prototypeJson); [](http://example.com/codeflow?start=12&length=71)

You could add an extra test here

FormulaValue prototypeValue2 = FormulaValueJSON.FromJson(prototypeJson, prototypeValue.Type); // with specified type

and verify 

Assert.Equal(prototypeValue.Type, prototypeValue2.Type);